src/modules/sensors/sensors.cpp

```diff
@@ -490,71 +501,102 @@ Sensors::adc_poll(struct sensor_combined_s &raw)
 								 ORB_PRIO_DEFAULT);
 					}
 
+#endif
+
+#ifdef ADC_AIRSPEED_VOLTAGE_CHANNEL
+#ifndef ADC_SONAR_VOLTAGE_CHANNEL
+
 				} else
 #endif
-				{
-					for (int b = 0; b < BOARD_NUMBER_BRICKS; b++) {
+#endif
 
-						/* Once we have subscriptions, Do this once for the lowest (highest priority
-						 * supply on power controller) that is valid.
+#ifdef ADC_SONAR_VOLTAGE_CHANNEL
+#ifdef ADC_AIRSPEED_VOLTAGE_CHANNEL
+
+				} else if (ADC_SONAR_VOLTAGE_CHANNEL == buf_adc[i].am_channel) {
+#else
+
+					if (ADC_SONAR_VOLTAGE_CHANNEL == buf_adc[i].am_channel) {
+#endif
+				float voltage_scaling = 2.04f;
+				float distance = buf_adc[i].am_data * 0.001f * voltage_scaling;
+
+				struct distance_sensor_s distance_report;
+				distance_report.timestamp = t;
+				distance_report.type = distance_sensor_s::MAV_DISTANCE_SENSOR_ULTRASOUND;
+				distance_report.orientation = 8;
```

Comment:
Again, shouldn't one be able to choose what orientation and sensor type is being imputed?